### PR TITLE
Fix guava cache configurations to respect cache eviction times.

### PR DESCRIPTION
**Pull request by wtritch**
Our previous guava cache configuration wasn't configuring the caches with eviction times.  The GuavaCacheManager ignores any registered Cache beans.

### CODE REVIEW

File: common/src/main/java/tds/common/cache/CacheConfigurationProperties.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package tds.common.cache;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ *
```
Comment (millerts) [inline common/src/main/java/tds/common/cache/CacheConfigurationProperties.java]: Let's remove this

File: common/pom.xml [hunk 2]
```diff
@@ -132,6 +132,17 @@
...
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (wtritch) [inline common/pom.xml:144]: Added some unit test dependencies.

**Review by millerts — APPROVED** [review]

File: common/src/main/java/tds/common/cache/CacheConfigurationProperties.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ */
+@ConfigurationProperties(prefix = "tds.cache")
+public class CacheConfigurationProperties {
+    private int expireTimeShort = 20;
+    private int expireTimeMedium = 600;
+    private int expireTimeLong = 7200;
```
Comment (wtritch) [inline common/src/main/java/tds/common/cache/CacheConfigurationProperties.java:12]: Move configuration values into a ConfigurationProperties class and renamed these configuration properties slightly from:
```
tds.cache.expire.time.short => tds.cache.expireTimeShort
```

File: common/src/main/java/tds/common/cache/NameAwareGuavaCacheManager.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+     * @param cacheNames        The cache names
+     */
+    public NameAwareGuavaCacheManager(final CacheConfigurationProperties cacheProperties,
+                                      final String... cacheNames) {
+        this.cacheProperties = cacheProperties;
+        this.setCacheNames(Arrays.asList(cacheNames));
+    }
+
+    @Override
+    protected Cache createGuavaCache(final String name) {
```
Comment (wtritch) [inline common/src/main/java/tds/common/cache/NameAwareGuavaCacheManager.java:32]: This method is called by the GuavaCacheManager when trying to find a non-existent cache by name.  The default implementation creates a basic Guava cache.  This override configures the created cache based upon the name.

File: common/src/main/java/tds/common/configuration/CacheConfiguration.java [hunk 5]
```diff
@@ -1,70 +1,43 @@
...
-        return new GuavaCache(CacheType.MEDIUM_TERM, CacheBuilder.newBuilder()
-            .expireAfterWrite(mediumTermExpirationInSeconds, TimeUnit.SECONDS)
-            .build());
-    }
-
-    @Bean
-    public Cache longTermCache() {
-        return new GuavaCache(CacheType.LONG_TERM, CacheBuilder.newBuilder()
-            .expireAfterWrite(longTermExpirationInSeconds, TimeUnit.SECONDS)
-            .build());
```
Comment (wtritch) [inline common/src/main/java/tds/common/configuration/CacheConfiguration.java:68]: These beans were ignored by the GuavaCacheManager.  It creates and stores its own Cache implementations by name.
The end result is that all our previous caching had no eviction times configured.

File: common/src/main/java/tds/common/cache/CacheConfigurationProperties.java (hunk 1 above)
Comment (wtritch) [reply common/src/main/java/tds/common/cache/CacheConfigurationProperties.java]: Added a class-level javadoc

**Pull request merged**